TestsExample/src/Test1032.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import { Alert, Button, Switch, Text, View } from 'react-native';
+import { NavigationContainer, ParamListBase } from '@react-navigation/native';
+import { createNativeStackNavigator, NativeStackNavigationProp } from 'react-native-screens/native-stack';
+
+const Stack = createNativeStackNavigator();
+
+export default function App() {
+    const [gestureEnabled, setGestureEnable] = React.useState(false);
+    return (
+        <View style={{ flex: 1, paddingBottom: 200 }}>
+            <NavigationContainer>
+                <Stack.Navigator screenOptions={{ gestureEnabled }}>
+                    <Stack.Screen name="Top" component={First} options={{ replaceAnimation: 'push' }} />
+                    <Stack.Screen name="Top1" component={Second} options={{ replaceAnimation: 'push' }} />
+                </Stack.Navigator>
+            </NavigationContainer >
+            <View>
+                <Text>Gesture enabled</Text>
+                <Switch value={gestureEnabled} onValueChange={setGestureEnable} />
```

Comment:
You can switch `gestureEnabled` directly in `Second` by adding such switch there and using `navigation.setOptions`.

Reply:
I think that being able to change it on both screens is useful, so I would keep it outside